MessageGeneration/MessageParser.cs

```diff
@@ -0,0 +1,452 @@
+﻿using Microsoft.CodeAnalysis.CSharp;
+
+namespace RosNet.MessageGeneration;
+
+public class MessageParser
+{
+
+    private readonly List<MessageToken> _tokens;
+
+    private readonly string _inFilePath;
+    private readonly string _inFileName;
+
+    private readonly string _rosPackageName;
+    private readonly string _className;
+    private readonly string _rosMsgName;
+
+    private readonly string _outPath;
+    private string _outFilePath;
+
+    private readonly HashSet<string> _imports = new();
+
+    private readonly Dictionary<string, string> _builtInTypeMapping;
+    private readonly Dictionary<string, string> _builtInTypesDefaultInitialValues;
+
+    private uint _lineNum = 1;
+
+    private readonly Dictionary<string, string> _symbolTable = new();
+    private readonly HashSet<string> _constants = new();
+    private readonly Dictionary<string, int> _arraySizes = new();
+
+    private string _body = "";
+
+    private readonly List<string> _warnings = new();
+
+    public MessageParser(List<MessageToken> tokens, string outPath, string rosPackageName, string type, Dictionary<string, string> builtInTypeMapping, Dictionary<string, string> builtInTypesDefaultInitialValues, string className = "", string rosMsgName = "")
+    {
+        this._tokens = tokens;
+
+        this._inFilePath = tokens[0].Content;
+        this._inFileName = Path.GetFileNameWithoutExtension(_inFilePath);
+
+        this._rosPackageName = rosPackageName;
+
+        this._className = className.Length == 0 ? Utilities.CapitalizeFirstLetter(_inFileName) : className;
+
+        this._rosMsgName = rosMsgName.Length == 0 ? Utilities.CapitalizeFirstLetter(_inFileName) : rosMsgName;
+
+
+        this._outPath = outPath;
+        this._outFilePath = Path.Combine(outPath, type);
+
+        this._tokens.RemoveAt(0);
+
+        this._builtInTypeMapping = builtInTypeMapping;
+        this._builtInTypesDefaultInitialValues = builtInTypesDefaultInitialValues;
+    }
+
+    public void Parse()
```

Comment:
1. This has a horrible name, as it generates code, not parse the `*.{msg,srv,action}`
2. This can be expanded with a custom interface / method such that we can support serialization of custom types.